src/chains.rs

```diff
@@ -254,10 +254,16 @@ impl ChainItemKind {
                     ChainItemKind::Parent {
                         expr: expr.clone(),
                         parens: is_method_call_receiver
-                            && matches!(
+                            && (matches!(
                                 &expr.kind,
                                 ast::ExprKind::Lit(lit) if crate::expr::lit_ends_in_dot(lit)
-                            ),
+                            ) || matches!(
+                                &expr.kind,
+                                ast::ExprKind::Closure(ref cl) if matches!(
+                                    cl.body.kind,
+                                    ast::ExprKind::Range(None, None, ast::RangeLimits::HalfOpen),
+                                )
+                            )),
```

Comment:
Could this be extracted to a method? I would prefer if this is pulled to a separate function that uses `match` instead of `matches`.